Source/DafnyCore/AST/Expressions/Operators/DecreasesToExpr.cs

```diff
@@ -0,0 +1,39 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.Dafny;
+
+public class DecreasesToExpr : Expression, ICloneable<DecreasesToExpr> {
```

Comment:
Are you using `DecreasesToExpr` also as a target for a `Rewriter`, to generate implicit `decreases` expressions? That might simplify code in `BoogieGenerator`